packages/features/timezone-buddy/components/AvailabilityEditSheet.tsx

```diff
@@ -0,0 +1,199 @@
+  });
+
+  const form = useForm<AvailabilityFormValues>({
+    values: data && {
+      ...data,
+      timeZone: data?.timeZone || Intl.DateTimeFormat().resolvedOptions().timeZone,
+      schedule: data?.availability || [],
+    },
+  });
+
+  const watchTimezone = form.watch("timeZone");
+
+  return (
+    <Sheet open={props.open} onOpenChange={props.onOpenChange}>
+      <Form
+        form={form}
+        id="availability-form"
+        handleSubmit={async ({ dateOverrides, ...values }) => {
+          // Just blocking on a UI side -> Backend will also do the validation
+          if (!hasEditPermission) return;
+          data &&
+            updateMutation.mutate({
+              scheduleId: data?.id,
+              dateOverrides: dateOverrides.flatMap((override) => override.ranges),
+              ...values,
+            });
+        }}>
+        <SheetContent
+          bottomActions={
+            <>
+              <Button color="secondary" className="w-full justify-center">
+                {t("cancel")}
+              </Button>
+              <Button
+                disabled={!hasEditPermission}
+                className="w-full justify-center"
+                type="submit"
+                loading={updateMutation.isLoading}
+                form="availability-form">
+                {t("save")}
+              </Button>
+            </>
+          }>
+          {!data?.hasDefaultSchedule && !isLoading && hasEditPermission && (
+            <div className="my-2">
+              <TopBanner
+                variant="warning"
+                text="This user has not completed onboarding. You will not be able to set their availability until they have completed onboarding. "
+              />
+            </div>
+          )}
+          {!hasEditPermission && !loadingPermissions && (
+            <div className="my-2">
+              <TopBanner
+                variant="warning"
+                text="You are viewing this user's availability. You can only edit your own availability."
+              />
+            </div>
+          )}
+
+          <SheetHeader>
+            <SheetTitle>Edit Users Availability : {props.selectedUser?.username}</SheetTitle>
+          </SheetHeader>
+
+          <>
+            <div>
+              <Label className="text-emphasis">
+                <>{t("timezone")}</>
+              </Label>
+              <TimezoneSelect
+                id="timezone"
+                isDisabled={!hasEditPermission}
+                value={watchTimezone ?? "Europe/London"}
+                onChange={(event) => {
+                  if (event) form.setValue("timeZone", event.value, { shouldDirty: true });
+                }}
+              />
+            </div>
+            <div className="mt-4">
+              <Label className="text-emphasis">{t("members_default_schedule")}</Label>
+              {/* Remove padding from schedule without touching the component */}
+              <div className="[&>*:first-child]:!p-0">
```

Comment:
Alex this was easier than modifying the schedule component - not the cleanest but it works